server/types/connection.go

```diff
@@ -2,13 +2,17 @@ package types
 
 import (
 	"context"
+	"net"
 
 	"github.com/open-telemetry/opamp-go/protobufs"
 )
 
 // Connection represents one OpAMP WebSocket connections.
 // The implementation MUST be a comparable type so that it can be used as a map key.
 type Connection interface {
```

Comment:
@andykellr to support the use case that you have I think we also need to add `net.Conn` as a field. This will allow to access the TLS certificate. I believe `net.Conn` is accessible for WebSocket. If you can confirm it is also accessible for gRPC connections and for plain HTTP connections then it should be OK to add it to this interface.

Reply:
I will go ahead and merge this PR. Feel free to submit a new PR that exposes net.Conn.

Reply:
@andykellr this PR looks good to me now. I will merge as is and you can address this comment in a separate PR if you need the `Conn`.